spec.bs

```diff
@@ -9278,28 +9362,117 @@ batcher=] |batcher| and a [=trusted scoring signals request=] |request|:
+
+1. If |input| [=list/is not empty=]:
+  1. [=list/Append=] |param| to |queryParamsList|.
+  1. Let |sizes| be an empty [=list=] of [=strings=].
+  1. [=list/For each=] |creative| of |input|:
+    1. If |creative|'s [=creative info/ad descriptor=]'s [=ad descriptor/size=] is null,
+      [=list/append=] "`,`" to |sizes|.
+    1. Otherwise, [=list/append=] the result of [=converting an ad size to a string=] given
+      |creative|'s [=creative info/ad descriptor=]'s [=ad descriptor/size=] to |sizes|.
+  1. [=list/Append=] the result of [=string/concatenating=] |sizes| with separator set to "," to
+    |queryParamsList|.
+
+  Note: this works differently from other query parameters since it has a comma that we do not
+  want to escape, and none of the other characters used to represent the width and height of each ad
+  need to be escaped either.
+
+</div>
+
+<div algorithm>
+To <dfn>build trusted scoring signals url</dfn> given a [=boolean=] |sendCreativeScanningMetadata|,
+a [=URL=] |signalsUrl|, an [=ordered set=] of [=creative info=] |ads|, an [=ordered set=] of
+[=creative info=] |componentAds|, an {{unsigned short}} |experimentGroupId|, and an [=origin=]
+|topLevelOrigin|:
+
+Note: When trusted scoring signals fetches are not batched, |ads|'s [=list/size=] is 1.
 
 1. Let |queryParamsList| be a new empty [=list=].
+
 1. [=list/Append=] "hostname=" to |queryParamsList|.
 1. [=list/Append=] the result of [=string/UTF-8 percent-encoding=] |topLevelOrigin| using
   [=component percent-encode set=] to |queryParamsList|.
-1. If |renderURLs| is not [=set/is empty|empty=]:
-  1. [=list/Append=] "`&renderURLs=`" to |queryParamsList|.
-  1. [=list/Extend=] |queryParamsList| with the result of [=encode trusted signals keys=] with
-    |renderURLs|.
-1. If |adComponentRenderURLs| is not [=set/is empty|empty=]:
-  1. [=list/Append=] "`&adComponentRenderURLs=`" to |queryParamsList|.
-  1. [=list/Extend=] |queryParamsList| with the result of [=encode trusted signals keys=] with
-    |adComponentRenderURLs|.
+1. Let |renderURLs| be the result of [=extracting URLs from creative info=] given |ads|.
+1. [=Add trusted signals param if needed=] given "`&renderURLs=`", |renderURLs|,
+  |queryParamsList|.
+1. Let |adComponentRenderURLs| be the result of [=extracting URLs from creative info=] given
+  |componentAds|.
+1. If |adComponentRenderURLs| is not [=list/is empty|empty=]:
+1. [=Add trusted signals param if needed=] given "`&adComponentRenderURLs=`",
+  |adComponentRenderURLs|, |queryParamsList|.
 1. If |experimentGroupId| is not null:
   1. [=list/Append=] "`&experimentGroupId=`" to |queryParamsList|.
   1. [=list/Append=] [=serialize an integer|serialized=] |experimentGroupId| to |queryParamsList|.
+1. If |sendCreativeScanningMetadata| is true:
+  1. [=Add trusted signals param if needed=] given "`&adCreativeScanningMetadata=`",
+    the result of [=extracting creative scanning metadata=] given |ads|, |queryParamsList|.

```

Comment:
nit: it's kinda confusing about which param is for which algorithm, but I don't see a better way to reword it.